scriptmodules/helpers.sh

```diff
@@ -131,6 +131,56 @@ function editFile() {
     [[ -n "$choice" ]] && echo "$choice" >"$file"
 }
 
+## @fn configKernelModule()
+## @brief mode configuration mode (install or remove)
+## @brief module module name to be added to /etc/modules-load.d file
+## @param parameter module parameter to be added to /etc/modprobe.d file
+## @param prefix prefix name of .conf files in /etc/modprobe.d & /etc/modules-load.d
+function configKernelModule() {
+    local mode="$1"
+    local module="$2"
+    local parameter="$3"
+    local prefix="$4"
+
+    [[ -z "$prefix" ]] && prefix="$module"
+
+    local parameter_file="/etc/modprobe.d/$prefix.conf"
+    local module_file="/etc/modules-load.d/$prefix.conf"
+
+    # SysVinit legacy compatibility
+    [[ ! -d "/etc/modules-load.d/" ]] && module_file="/etc/modules"
+
+    if [[ "$mode" == "remove" ]]; then
+        # remove all entries matching module name from configs
+        sed -i "/$module\b/d" "$module_file" "$parameter_file" 2>/dev/null
+        # delete empty config files (if necessary)
+        [[ ! -s "$module_file" && -d "/etc/modules-load.d/" ]] && rm -f "$module_file"
+        [[ ! -s "$parameter_file" ]] && rm -f "$parameter_file"
+        # rmmod module (if necessary)
+        [[ -n "$(lsmod | grep $module)" ]] && rmmod "$module"
+        return
+    fi
+
+    # add parameter to config
+    if ! grep -qs "$parameter" "$parameter_file" && [[ -n "$parameter" ]]; then
```

Comment:
if `[[ -n parameter` would be better first. else you do the grep even if the parameter is empty.